table/block_based/filter_policy.cc

```diff
@@ -1311,6 +1294,42 @@ Status XXPH3FilterBitsBuilder::MaybePostVerify(const Slice& filter_content) {
 }
 }  // namespace
 
+Slice FilterBitsBuilder::Finish(std::unique_ptr<const char[]>* buf) {
+  Slice filter;
+  Status s = FinishV2(/*allocator*/ nullptr, &filter);
+  if (filter.size() > 0) {
+    assert(s.ok());
+    buf->reset(filter.data());
+    return filter;
+  } else if (s.ok()) {
+    // Empty filter
+    return Slice();
+  } else {
+    // Safe fallback
+    return kAlwaysTrueFilter;
+  }
+}
+
+Status FilterBitsBuilder::FinishV2(MemoryAllocator* allocator,
+                                   Slice* output_filter) {
+  std::unique_ptr<const char[]> buf;
+  Slice slice = Finish(&buf);
+
+  if (allocator == nullptr && buf.get() == slice.data()) {
```

Comment:
I wonder if we should assert(buf.get() == slice.data()) as a constant truth since we mentioned "// The ownership of actual data is set to buf." in the API of FilterBitsBuilder::Finish(). Then we split the cases into if(allocator == nullptr){..}else{}. I am not sure why we specifically handled the case `slice.empty()`, given that the `Finish(&buf)` get called here is a customized function that we don't know much about (other than the API doc/requirement of FilterBitsBuilder::Finish())